# UDEMY COURSE DATASET ANALYSIS

In this project , I am going to anlayze the udemy dataset available in CSV file taken from kaggle using pandas DataFrame

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv(r"C:\Users\utsha\OneDrive\Desktop\Data analysis project\dataset\7. Udemy Courses.csv")

In [3]:
df

course_id                                       course_title  is_paid  \
0        288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1       1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   
2       1193886  #12 Hand Coordination: Let your Hands dance wi...     True   
3       1116700  #4 Piano Hand Coordination: Fun Piano Runs in ...     True   
4       1120410  #5  Piano Hand Coordination:  Piano Runs in 2 ...     True   
...         ...                                                ...      ...   
3677     328960  Your Own Site in 45 Min: The Complete Wordpres...     True   
3678     552700    Your Second Course on Piano: Two Handed Playing     True   
3679     631754  Zend Framework 2: Learn the PHP framework ZF2 ...     True   
3680    1225194  Zoho Books Gestion Financière d'Entreprise pas...    False   
3681     964478                Zombie Apocalypse Photoshop Actions     True   

     price  num_subscribers  num_reviews  num_lectures               level  \
0       35             3137           18            68          All Levels   
1       75             1593            1            41  Intermediate Level   
2       75              482            1            47  Intermediate Level   
3       75              850            3            43  Intermediate Level   
4       75              940            3            32  Intermediate Level   
...    ...              ...          ...           ...                 ...   
3677   120             1566           29            36          All Levels   
3678    70             1018           12            22      Beginner Level   
3679    40              723          130            37          All Levels   
3680  Free              229            0            33          All Levels   
3681    50               12            1            15          All Levels   

     content_duration   published_timestamp              subject  
0           1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1              1 hour  2017-04-12T19:06:34Z  Musical Instruments  
2           1.5 hours  2017-04-26T18:34:57Z  Musical Instruments  
3              1 hour  2017-02-21T23:48:18Z  Musical Instruments  
4             37 mins  2017-02-21T23:44:49Z  Musical Instruments  
...               ...                   ...                  ...  
3677          4 hours  2015-04-20T22:15:17Z      Web Development  
3678          5 hours  2015-10-26T20:04:21Z  Musical Instruments  
3679        6.5 hours  2015-11-11T18:55:45Z      Web Development  
3680          2 hours  2017-05-26T16:45:55Z     Business Finance  
3681        1.5 hours  2016-09-26T22:19:48Z       Graphic Design  

[3682 rows x 11 columns]

In [4]:
df.isnull().sum()

course_id              0
course_title           0
is_paid                0
price                  0
num_subscribers        0
num_reviews            0
num_lectures           0
level                  0
content_duration       0
published_timestamp    0
subject                0
dtype: int64

There are no null values in datas


# 1) What are all different subjects for which Udemy is offering courses ?

In [5]:
df.head()

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   
2    1193886  #12 Hand Coordination: Let your Hands dance wi...     True   
3    1116700  #4 Piano Hand Coordination: Fun Piano Runs in ...     True   
4    1120410  #5  Piano Hand Coordination:  Piano Runs in 2 ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   
2    75              482            1            47  Intermediate Level   
3    75              850            3            43  Intermediate Level   
4    75              940            3            32  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments  
2        1.5 hours  2017-04-26T18:34:57Z  Musical Instruments  
3           1 hour  2017-02-21T23:48:18Z  Musical Instruments  
4          37 mins  2017-02-21T23:44:49Z  Musical Instruments

In [6]:
df.subject.unique()

array(['Musical Instruments', 'Business Finance', 'Graphic Design',
       'Web Development'], dtype=object)

### Udemy is offering course in 4 different subject and they are Musical Instruments,Business Finance,Graphic Design,web Development.


#  2) Which subject has the maximum number of courses.

In [7]:
df.head(2)

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments

In [8]:
df.subject.value_counts()

Web Development        1200
Business Finance       1199
Musical Instruments     680
Graphic Design          603
Name: subject, dtype: int64

# 3) Show all the courses which are Free of Cost.

In [9]:
df.head(2)

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments

In [10]:
df[df.is_paid == False]

course_id                                       course_title  is_paid  \
41       286070        5 lecciones que todo guitarrista debe tomar    False   
49       696630  7 Ways A Beginner Guitarist Can Sound Better, ...    False   
57       955914  A beginner`s guide to fingerpicking and strumm...    False   
67       270976                             A how to guide in HTML    False   
74      1214144  ¡Triunfar en La Bolsa de Valores No Requiere d...    False   
...         ...                                                ...      ...   
3620     552598  WordPress in under an hour - Create/manage you...    False   
3638    1068590                     Wordpress Website Mastery 2017    False   
3652     752706  Write quicker HTML5 and CSS 3; productivity ha...    False   
3674    1257976  Your First 10 Guitar Lessons - Learn how to pl...    False   
3680    1225194  Zoho Books Gestion Financière d'Entreprise pas...    False   

     price  num_subscribers  num_reviews  num_lectures           level  \
41    Free             4452          263            14  Beginner Level   
49    Free             4529          193             7  Beginner Level   
57    Free             3481           29            20  Beginner Level   
67    Free             7318          205             8  Beginner Level   
74    Free              338            7             6  Beginner Level   
...    ...              ...          ...           ...             ...   
3620  Free             6474           86            15      All Levels   
3638  Free             5181           64            22  Beginner Level   
3652  Free            13016          286            15      All Levels   
3674  Free              924            4            18  Beginner Level   
3680  Free              229            0            33      All Levels   

     content_duration   published_timestamp              subject  
41             1 hour  2014-08-23T05:08:14Z  Musical Instruments  
49            36 mins  2015-12-21T18:50:50Z  Musical Instruments  
57            2 hours  2016-09-13T21:51:59Z  Musical Instruments  
67            35 mins  2014-08-10T20:19:10Z      Web Development  
74             1 hour  2017-05-30T14:30:12Z     Business Finance  
...               ...                   ...                  ...  
3620           1 hour  2015-07-15T00:21:56Z      Web Development  
3638        2.5 hours  2017-01-18T17:01:40Z      Web Development  
3652        1.5 hours  2016-03-03T17:45:31Z      Web Development  
3674           1 hour  2017-06-29T00:29:36Z  Musical Instruments  
3680          2 hours  2017-05-26T16:45:55Z     Business Finance  

[310 rows x 11 columns]

310 courses are free 

# 4) Show all the courses which are Paid.

In [11]:
df.head(2)

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments

In [12]:
df[df.is_paid == True]

course_id                                       course_title  is_paid  \
0        288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1       1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   
2       1193886  #12 Hand Coordination: Let your Hands dance wi...     True   
3       1116700  #4 Piano Hand Coordination: Fun Piano Runs in ...     True   
4       1120410  #5  Piano Hand Coordination:  Piano Runs in 2 ...     True   
...         ...                                                ...      ...   
3676     498488  Your First Successful Forex Trades - With Case...     True   
3677     328960  Your Own Site in 45 Min: The Complete Wordpres...     True   
3678     552700    Your Second Course on Piano: Two Handed Playing     True   
3679     631754  Zend Framework 2: Learn the PHP framework ZF2 ...     True   
3681     964478                Zombie Apocalypse Photoshop Actions     True   

     price  num_subscribers  num_reviews  num_lectures               level  \
0       35             3137           18            68          All Levels   
1       75             1593            1            41  Intermediate Level   
2       75              482            1            47  Intermediate Level   
3       75              850            3            43  Intermediate Level   
4       75              940            3            32  Intermediate Level   
...    ...              ...          ...           ...                 ...   
3676   200             1079           34            16          All Levels   
3677   120             1566           29            36          All Levels   
3678    70             1018           12            22      Beginner Level   
3679    40              723          130            37          All Levels   
3681    50               12            1            15          All Levels   

     content_duration   published_timestamp              subject  
0           1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1              1 hour  2017-04-12T19:06:34Z  Musical Instruments  
2           1.5 hours  2017-04-26T18:34:57Z  Musical Instruments  
3              1 hour  2017-02-21T23:48:18Z  Musical Instruments  
4             37 mins  2017-02-21T23:44:49Z  Musical Instruments  
...               ...                   ...                  ...  
3676        2.5 hours  2015-05-26T20:48:48Z     Business Finance  
3677          4 hours  2015-04-20T22:15:17Z      Web Development  
3678          5 hours  2015-10-26T20:04:21Z  Musical Instruments  
3679        6.5 hours  2015-11-11T18:55:45Z      Web Development  
3681        1.5 hours  2016-09-26T22:19:48Z       Graphic Design  

[3372 rows x 11 columns]

lots of course

# arranging course in term of price

In [13]:
df.head()

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   
2    1193886  #12 Hand Coordination: Let your Hands dance wi...     True   
3    1116700  #4 Piano Hand Coordination: Fun Piano Runs in ...     True   
4    1120410  #5  Piano Hand Coordination:  Piano Runs in 2 ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   
2    75              482            1            47  Intermediate Level   
3    75              850            3            43  Intermediate Level   
4    75              940            3            32  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments  
2        1.5 hours  2017-04-26T18:34:57Z  Musical Instruments  
3           1 hour  2017-02-21T23:48:18Z  Musical Instruments  
4          37 mins  2017-02-21T23:44:49Z  Musical Instruments

In [14]:
    df.sort_values(by = 'price',ascending = False)

course_id                                       course_title  is_paid  \
3117     357726  Spring Framework 4 And Dependency Injection Fo...    False   
296     1127988                     Angular Tutorial For Beginners    False   
2043     297390  Joomla: Create a Joomla Website This Weekend W...    False   
311      692176  Apply finance concepts for smart project manag...    False   
2434     777654                          Learn what's new in PHP 7    False   
...         ...                                                ...      ...   
3510     576694                   Ultimate WordPress Plugin Course     True   
1227    1136912  Financial Accounting - The Ultimate Beginner C...     True   
2161     511378                Learn CSS Transition and Animation      True   
1198    1064904                     Evolution and Features of ETFs     True   
3156     380206           Stock Trading Essentials for Day Traders     True   

     price  num_subscribers  num_reviews  num_lectures               level  \
3117  Free            21147         1220            16      Beginner Level   
296   Free             5397          112            36      Beginner Level   
2043  Free            18496          386            30      Beginner Level   
311   Free             3465          161            24      Beginner Level   
2434  Free            12336          271            26  Intermediate Level   
...    ...              ...          ...           ...                 ...   
3510   100             4537          415            49          All Levels   
1227   100                9            2           101      Beginner Level   
2161   100             3757           28            20          All Levels   
1198   100                0            0            10          All Levels   
3156   100               94           15             9          All Levels   

     content_duration   published_timestamp           subject  
3117        1.5 hours  2015-01-16T07:45:28Z   Web Development  
296         2.5 hours  2017-02-28T04:07:50Z   Web Development  
2043          2 hours  2014-10-06T20:19:00Z   Web Development  
311           2 hours  2016-02-23T21:05:46Z  Business Finance  
2434        3.5 hours  2016-04-04T22:32:13Z   Web Development  
...               ...                   ...               ...  
3510          9 hours  2015-08-20T20:32:31Z   Web Development  
1227         10 hours  2017-04-06T00:39:12Z  Business Finance  
2161           1 hour  2015-05-27T17:51:19Z   Web Development  
1198          43 mins  2017-04-03T22:17:10Z  Business Finance  
3156           1 hour  2015-01-08T17:40:22Z  Business Finance  

[3682 rows x 11 columns]

#  5) Which are Top Selling Courses ?

In [15]:
df.head()

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   
2    1193886  #12 Hand Coordination: Let your Hands dance wi...     True   
3    1116700  #4 Piano Hand Coordination: Fun Piano Runs in ...     True   
4    1120410  #5  Piano Hand Coordination:  Piano Runs in 2 ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   
2    75              482            1            47  Intermediate Level   
3    75              850            3            43  Intermediate Level   
4    75              940            3            32  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments  
2        1.5 hours  2017-04-26T18:34:57Z  Musical Instruments  
3           1 hour  2017-02-21T23:48:18Z  Musical Instruments  
4          37 mins  2017-02-21T23:44:49Z  Musical Instruments

In [16]:
df.sort_values(by = 'num_subscribers' , ascending = False)

course_id                                       course_title  is_paid  \
2230      41295               Learn HTML5 Programming From Scratch    False   
776       59014                     Coding for Entrepreneurs Basic    False   
3385     625204                         The Web Developer Bootcamp     True   
640      173548  Build Your First Website in 1 Week with HTML5 ...    False   
3316     764164              The Complete Web Developer Course 2.0     True   
...         ...                                                ...      ...   
2290    1223240  Learn Pirates of the Caribbean by Ear on the P...     True   
1267    1258666                Financial Statement Auditing Cycles     True   
2070    1215926                Kickstarter success in 5 easy steps     True   
1931    1247992  Introduction to Project Management for Finance...     True   
3165    1087466  Stop Creditors from Harassing you and Avoid Ba...     True   

     price  num_subscribers  num_reviews  num_lectures               level  \
2230  Free           268923         8629            45          All Levels   
776   Free           161029          279            27      Beginner Level   
3385   200           121584        27445           342          All Levels   
640   Free           120291         5924            30      Beginner Level   
3316   200           114512        22412           304          All Levels   
...    ...              ...          ...           ...                 ...   
2290    20                0            0             6          All Levels   
1267    50                0            0             9  Intermediate Level   
2070    20                0            0            12          All Levels   
1931    50                0            0             9      Beginner Level   
3165    20                0            0             7      Beginner Level   

     content_duration   published_timestamp              subject  
2230       10.5 hours  2013-02-14T07:03:41Z      Web Development  
776         3.5 hours  2013-06-09T15:51:55Z      Web Development  
3385         43 hours  2015-11-02T21:13:27Z      Web Development  
640           3 hours  2014-04-08T16:21:30Z      Web Development  
3316       30.5 hours  2016-03-08T22:28:36Z      Web Development  
...               ...                   ...                  ...  
2290          32 mins  2017-05-22T17:14:43Z  Musical Instruments  
1267          2 hours  2017-06-29T23:20:10Z     Business Finance  
2070          31 mins  2017-05-16T14:55:28Z     Business Finance  
1931          2 hours  2017-07-03T21:40:32Z     Business Finance  
3165          37 mins  2017-02-02T16:22:37Z     Business Finance  

[3682 rows x 11 columns]

#  6) Which are Least Selling Courses ?

In [17]:
df.sort_values(by = 'num_subscribers')

course_id                                       course_title  is_paid  \
649     1233314                      Building a Balanced Scorecard     True   
3259    1232282          The Cash Flow Statement - An Introduction     True   
3261    1275872                    The Complete Adobe Spark Course     True   
910      726314       Create Beautiful Image Maps for Your Website     True   
3328     715476        The Cuckoo Strategy to get European Funding     True   
...         ...                                                ...      ...   
3316     764164              The Complete Web Developer Course 2.0     True   
640      173548  Build Your First Website in 1 Week with HTML5 ...    False   
3385     625204                         The Web Developer Bootcamp     True   
776       59014                     Coding for Entrepreneurs Basic    False   
2230      41295               Learn HTML5 Programming From Scratch    False   

     price  num_subscribers  num_reviews  num_lectures               level  \
649     50                0            0            11  Intermediate Level   
3259    50                0            0            10      Beginner Level   
3261   150                0            0            29          All Levels   
910     20                0            0             8  Intermediate Level   
3328   160                0            0             6          All Levels   
...    ...              ...          ...           ...                 ...   
3316   200           114512        22412           304          All Levels   
640   Free           120291         5924            30      Beginner Level   
3385   200           121584        27445           342          All Levels   
776   Free           161029          279            27      Beginner Level   
2230  Free           268923         8629            45          All Levels   

     content_duration   published_timestamp           subject  
649           2 hours  2017-07-03T21:38:22Z  Business Finance  
3259        1.5 hours  2017-06-28T16:05:51Z  Business Finance  
3261          2 hours  2017-07-03T17:39:57Z    Graphic Design  
910           37 mins  2016-01-18T17:56:36Z    Graphic Design  
3328           1 hour  2016-01-06T02:38:23Z  Business Finance  
...               ...                   ...               ...  
3316       30.5 hours  2016-03-08T22:28:36Z   Web Development  
640           3 hours  2014-04-08T16:21:30Z   Web Development  
3385         43 hours  2015-11-02T21:13:27Z   Web Development  
776         3.5 hours  2013-06-09T15:51:55Z   Web Development  
2230       10.5 hours  2013-02-14T07:03:41Z   Web Development  

[3682 rows x 11 columns]

#  7) Show all courses of Graphic Design where the price is below 100 ?

In [18]:
df.head(2)

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments

In [19]:
df.subject.unique()

array(['Musical Instruments', 'Business Finance', 'Graphic Design',
       'Web Development'], dtype=object)

In [20]:
df[(df['subject'] == 'Graphic Design') & (df['price'] < '100')]

Empty DataFrame
Columns: [course_id, course_title, is_paid, price, num_subscribers, num_reviews, num_lectures, level, content_duration, published_timestamp, subject]
Index: []

In [21]:
df[df['subject'] == 'Graphic Design'].subject.value_counts()

Graphic Design    603
Name: subject, dtype: int64

There are 603 graphic designing course
but no course for less than 100

# 8) List out all the courses that are related to 'Python'.

In [22]:
df.head()

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   
2    1193886  #12 Hand Coordination: Let your Hands dance wi...     True   
3    1116700  #4 Piano Hand Coordination: Fun Piano Runs in ...     True   
4    1120410  #5  Piano Hand Coordination:  Piano Runs in 2 ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   
2    75              482            1            47  Intermediate Level   
3    75              850            3            43  Intermediate Level   
4    75              940            3            32  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments  
2        1.5 hours  2017-04-26T18:34:57Z  Musical Instruments  
3           1 hour  2017-02-21T23:48:18Z  Musical Instruments  
4          37 mins  2017-02-21T23:44:49Z  Musical Instruments

In [23]:
df[df.course_title.str.contains('Python')]

course_id                                       course_title  is_paid  \
216      599504  Advanced Scalable Python Web Development Using...     True   
777       47963  Coding for Entrepreneurs: Learn Python, Django...     True   
814      631128  Complete Python Web Course: Build 8 Python Web...     True   
842      186096  Core: A Web App Reference Guide for Django, Py...     True   
1400     394832  Fun and creative web engineering with Python a...    False   
1932    1038538            Introduction to QGIS Python Programming     True   
1969     902888          Investment Portfolio Analysis with Python     True   
2294     391546        Learn Python and Django: Payment Processing     True   
2295    1063722            Learn Python Django - A Hands-On Course     True   
2341     546848  Learn to code in Python and learn Adobe Photos...     True   
2906     368340    Professional Python Web Development Using Flask     True   
2908    1035940  Professional RESTful API Design using Python F...     True   
2929     270808                      Projects in Django and Python     True   
2952    1170894  Python Algo Stock Trading: Automate Your Trading!     True   
2953    1070886         Python Algo Trading: FX Trading with Oanda     True   
2954    1196544   Python Algo Trading: Sentiment Trading with News     True   
2955     822444  Python and Django Full Stack Web Developer Boo...     True   
2956     477702  Python for Beginners: Python Programming Langu...     True   
2957    1035472  Python for Finance: Investment Fundamentals & ...     True   
2958     529828                     Python for Trading & Investing     True   
2959      70640  Python Programming: Create an Digital Marketpl...     True   
2960     523312                             Python Web Programming     True   
2962    1088656          Quantitative Trading Analysis with Python     True   
3013     970600                    REST APIs with Flask and Python     True   
3154     815482               Stock Technical Analysis with Python     True   
3272     938560  The Complete Ethical Hacking Course 2.0: Pytho...     True   
3460      76052  Try Django 1.9 | Build a Blog and Learn Python...     True   
3573      16646                        Web Programming with Python     True   
3574     574082        Web Scraping with Python, Ruby & import. io     True   

     price  num_subscribers  num_reviews  num_lectures               level  \
216    120             1299           56            71  Intermediate Level   
777    195            23412          799           251          All Levels   
814    110             7489          941           173          All Levels   
842    195             2497           98           154          All Levels   
1400  Free            10917          319            25          All Levels   
1932    85              197           26            28      Beginner Level   
1969    50              209           13            37          All Levels   
2294    70            17714          198            23          All Levels   
2295    50             1339           21            18      Beginner Level   
2341    50             1132            1            29          All Levels   
2906   120             3420          489           102      Beginner Level   
2908   120              578           25            36  Intermediate Level   
2929    60             1764           53            28          All Levels   
2952    95             1165           21            41      Beginner Level   
2953   200              453           42            33  Intermediate Level   
2954   200              294           19            42          All Levels   
2955   200            11832         1883           191          All Levels   
2956   150             6153          125            84      Beginner Level   
2957   195             3811          278           103          All Levels   
2958    95              638           25            36          All Levels 

# 9) What are courses that were published in the year 2015 ?

In [24]:
df.head(2)

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   

  content_duration   published_timestamp              subject  
0        1.5 hours  2014-09-18T05:07:05Z  Musical Instruments  
1           1 hour  2017-04-12T19:06:34Z  Musical Instruments

In [25]:
df.dtypes


course_id               int64
course_title           object
is_paid                  bool
price                  object
num_subscribers         int64
num_reviews             int64
num_lectures            int64
level                  object
content_duration       object
published_timestamp    object
subject                object
dtype: object

In [26]:
df.published_timestamp.dtypes

dtype('O')

Changing object datatypes to datetime

In [27]:
df['published_timestamp'] = pd.to_datetime(df['published_timestamp'])

In [28]:
df.published_timestamp.dtypes

datetime64[ns, UTC]

now we wil extract year and make it a new col

In [29]:
df['year'] = df.published_timestamp.dt.year

In [30]:
df.head(2)

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   

  content_duration       published_timestamp              subject  year  
0        1.5 hours 2014-09-18 05:07:05+00:00  Musical Instruments  2014  
1           1 hour 2017-04-12 19:06:34+00:00  Musical Instruments  2017

In [32]:
df[df.year == 2015]

course_id                                       course_title  is_paid  \
10       591880  1 - Concepts of Statistics For Beginners Step ...     True   
16       302450        10 Numbers Every Business Owner Should Know     True   
19       384928  101 Blues riffs - learn how the harmonica supe...     True   
23       550842  16 Guitar Chords to Jam With - (Beginner - Int...     True   
26       486240      2 Easy Steps To Investment And Avoiding Traps     True   
...         ...                                                ...      ...   
3675     551170                         Your First Course on Piano     True   
3676     498488  Your First Successful Forex Trades - With Case...     True   
3677     328960  Your Own Site in 45 Min: The Complete Wordpres...     True   
3678     552700    Your Second Course on Piano: Two Handed Playing     True   
3679     631754  Zend Framework 2: Learn the PHP framework ZF2 ...     True   

     price  num_subscribers  num_reviews  num_lectures               level  \
10     200              273            4            15      Beginner Level   
16      20               13            1             9          All Levels   
19     200             1350           65            55  Intermediate Level   
23      20             1224           19            20      Beginner Level   
26      20              828            1            20          All Levels   
...    ...              ...          ...           ...                 ...   
3675    70             2002           34            33      Beginner Level   
3676   200             1079           34            16          All Levels   
3677   120             1566           29            36          All Levels   
3678    70             1018           12            22      Beginner Level   
3679    40              723          130            37          All Levels   

     content_duration       published_timestamp              subject  year  
10            31 mins 2015-08-30 22:48:34+00:00     Business Finance  2015  
16             1 hour 2015-03-08 19:11:24+00:00     Business Finance  2015  
19          6.5 hours 2015-01-04 21:14:31+00:00  Musical Instruments  2015  
23             1 hour 2015-07-10 19:53:56+00:00  Musical Instruments  2015  
26             1 hour 2015-04-27 23:18:59+00:00     Business Finance  2015  
...               ...                       ...                  ...   ...  
3675          4 hours 2015-07-16 17:07:49+00:00  Musical Instruments  2015  
3676        2.5 hours 2015-05-26 20:48:48+00:00     Business Finance  2015  
3677          4 hours 2015-04-20 22:15:17+00:00      Web Development  2015  
3678          5 hours 2015-10-26 20:04:21+00:00  Musical Instruments  2015  
3679        6.5 hours 2015-11-11 18:55:45+00:00      Web Development  2015  

[1014 rows x 12 columns]

# 10) What is the Max. Number of Subscribers for Each Level of courses ?

In [34]:
df.head()

course_id                                       course_title  is_paid  \
0     288942  #1 Piano Hand Coordination: Play 10th Ballad i...     True   
1    1170074  #10 Hand Coordination - Transfer Chord Ballad ...     True   
2    1193886  #12 Hand Coordination: Let your Hands dance wi...     True   
3    1116700  #4 Piano Hand Coordination: Fun Piano Runs in ...     True   
4    1120410  #5  Piano Hand Coordination:  Piano Runs in 2 ...     True   

  price  num_subscribers  num_reviews  num_lectures               level  \
0    35             3137           18            68          All Levels   
1    75             1593            1            41  Intermediate Level   
2    75              482            1            47  Intermediate Level   
3    75              850            3            43  Intermediate Level   
4    75              940            3            32  Intermediate Level   

  content_duration       published_timestamp              subject  year  
0        1.5 hours 2014-09-18 05:07:05+00:00  Musical Instruments  2014  
1           1 hour 2017-04-12 19:06:34+00:00  Musical Instruments  2017  
2        1.5 hours 2017-04-26 18:34:57+00:00  Musical Instruments  2017  
3           1 hour 2017-02-21 23:48:18+00:00  Musical Instruments  2017  
4          37 mins 2017-02-21 23:44:49+00:00  Musical Instruments  2017

In [35]:
df.level.unique()

array(['All Levels', 'Intermediate Level', 'Beginner Level',
       'Expert Level'], dtype=object)

In [37]:
df.groupby('level')['num_subscribers'].max()

level
All Levels            268923
Beginner Level        161029
Expert Level            5172
Intermediate Level     29167
Name: num_subscribers, dtype: int64